src/devices/digital_input.rs

```diff
@@ -38,19 +35,107 @@ pub struct DigitalInput {
     detach_cb: Option<*mut c_void>,
 }
 
+/// InputMode for digital input http://perk-software.cs.queensu.ca/plus/doc/nightly/dev/phidget22_8h.html#a5ad0740978daad6539d3a8249607bd46
+#[derive(Copy, Clone)]
+pub enum InputMode {
+    /// NPN: cannot find docs
+    NPN,
+    /// PNP: cannot find docs
+    PNP,
+}
+
+/// PowerSupply for digital input http://perk-software.cs.queensu.ca/plus/doc/nightly/dev/phidget22_8h.html#a0293d3a21e8de247c4b562ceda897876
+#[derive(Copy, Clone)]
+pub enum PowerSupply {
+    /// OFF: cannot find docs
+    OFF,
+    /// V12: cannot find docs
+    V12,
+    /// v24: cannot find docs
+    V24,
+}
+
 impl DigitalInput {
     /// Create a new digital input.
     pub fn new() -> Self {
-        let mut chan: DigitalInputHandle = ptr::null_mut();
+        let mut chan: PhidgetDigitalInputHandle = ptr::null_mut();
         unsafe {
             ffi::PhidgetDigitalInput_create(&mut chan);
         }
         Self::from(chan)
     }
 
+    /// Set input mode
+    pub fn set_input_mode(&self, input_mode: InputMode) -> Result<()> {
+        let _v = match input_mode {
+            InputMode::NPN => 1,
+            InputMode::PNP => 0,
+        };
+
+        ReturnCode::result(unsafe { ffi::PhidgetDigitalInput_setInputMode(self.chan, _v) })?;
+        Ok(())
+    }
+
+    /// Get input mode
+    pub fn get_input_mode(&self) -> Result<InputMode> {
+        let mut _im: ffi::Phidget_InputMode = 0;
```

Comment:
Don't use a leading underscore if you are using the variable. Just `im`, not `_im`.

Reply:
ok